test/test_dataframe.py

```diff
@@ -1,13 +1,18 @@
 # Copyright (c) Facebook, Inc. and its affiliates.
+import os
 import unittest
+import warnings
+from itertools import chain
 
 import expecttest
+from _utils._common_utils_for_test import create_temp_dir, reset_after_n_next_calls
 
-from _utils._common_utils_for_test import reset_after_n_next_calls
+from torchdata.datapipes.iter import DataFrameMaker, FileLister, IterableWrapper, ParquetDFReader
 
-from torchdata.datapipes.iter import DataFrameMaker, IterableWrapper
 
 try:
+    import pyarrow
+    import pyarrow.parquet as parquet
```

Comment:
This is incorrect. You need to separate `pyarrow` and `torcharrow` into two try-except. If `pyarrow` doesn't exist and `torcharrow` exists, you will skip the test of `test_dataframe_maker_iterdatapipe`

Reply:
I was planning to skip over all tests if one of them is missing but I agree that keeping them separate is better.